src/popup/utils/utils.ts

```diff
@@ -33,28 +41,56 @@ export function formatTime(time: number) {
   });
 }
 
-export function checkHashType(fullHash: string) {
+export function getHashPrefix(fullHash: string) {
   const addressTypes: Record<string, string> = {
-    ak: 'account/transactions',
-    ct: 'contracts/transactions',
-    nm: 'names',
-    ok: 'oracles/queries',
-    th: 'transactions',
+    [HASH_PREFIX_ACCOUNT]: 'account/transactions',
+    [HASH_PREFIX_CONTRACT]: 'contracts/transactions',
+    [HASH_PREFIX_NAME]: 'names',
+    [HASH_PREFIX_ORACLE]: 'oracles/queries',
+    [HASH_PREFIX_TRANSACTION]: 'transactions',
   };
+
   const [prefix, hash] = fullHash.split('_');
 
-  let valid: boolean = false;
-  let endpoint: string | null = null;
+  return {
+    hash,
+    prefix,
+    addressTypes,
+  };
+}
 
-  if (fullHash.endsWith('.chain')) {
-    valid = true;
-    endpoint = addressTypes.nm;
-  } else if (addressTypes[prefix] && HASH_REGEX.test(hash)) {
-    valid = true;
-    endpoint = addressTypes[prefix];
+export function validateHash(fullHash?: string) {
+  if (!fullHash) {
+    return {};
```

Comment:
This function should always return the same structure. Returning empty object breaks this because when this function is used you can't be sure what will be returned.